pkg/server/storage_api/storage_api_test: TestHealthAPI failed

pkg/server/storage_api/storage_api_test.TestHealthAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20577856?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20577856?buildTab=artifacts#/) on master @ [bb1c098ecc7058a2b24a23820c5a8e235dfa714c](https://github.com/cockroachdb/cockroach/commits/bb1c098ecc7058a2b24a23820c5a8e235dfa714c):


```
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: status: 401 Unauthorized, content-type: text/plain; charset=utf-8, body: a valid authentication cookie or JWT is required
        , error: <nil>
    health_test.go:103: observed error: Get "https://127.0.0.1:33635/_admin/v1/health?ready=1": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    health_test.go:103: observed error: Get "https://127.0.0.1:33635/_admin/v1/health?ready=1": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    health_test.go:103: observed error: Get "https://127.0.0.1:33635/_admin/v1/health?ready=1": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    health_test.go:103: observed error: Get "https://127.0.0.1:33635/_admin/v1/health?ready=1": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    health_test.go:97: condition failed to evaluate within 45s: Get "https://127.0.0.1:33635/_admin/v1/health?ready=1": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
--- FAIL: TestHealthAPI/liveness (49.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHealthAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55257

--- a/pkg/server/server_http.go
+++ b/pkg/server/server_http.go
@@ -237,7 +237,7 @@ func (s *httpServer) setupRoutes(
 
 	// Exempt the 2nd health check endpoint from authentication.
 	// (This simply mirrors /health and exists for backward compatibility.)
-	s.mux.Handle(apiconstants.AdminHealth, authenticatedAPIInternalServer)
+	s.mux.Handle(apiconstants.AdminHealth, unauthenticatedAPIInternalServer)
 	// The /_status/vars and /metrics endpoint is not authenticated either. Useful for monitoring.
 	s.mux.Handle(apiconstants.StatusVars, http.HandlerFunc(varsHandler{metricSource, s.cfg.Settings, false /* useStaticLabels */}.handleVars))
 	s.mux.Handle(apiconstants.MetricsPath, http.HandlerFunc(varsHandler{metricSource, s.cfg.Settings, true /* useStaticLabels */}.handleVars))

--- a/pkg/server/storage_api/health_test.go
+++ b/pkg/server/storage_api/health_test.go
@@ -18,7 +18,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/server/srvtestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -30,7 +29,6 @@ func TestHealthAPI(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 155052)
 	ctx := context.Background()
 
 	t.Run("sql", func(t *testing.T) {

